rocksdb_admin/admin_handler.cpp

```diff
@@ -929,6 +974,177 @@ void AdminHandler::async_tm_addS3SstFilesToDB(
   callback->result(AddS3SstFilesToDBResponse());
 }
 
+int AdminHandler::getPartitionIDFromDBName(const std::string& db_name) {
```

Comment:
this might not work when a segment name contains numbers. Such as p2p1, p2p2, etc.

We can do something similar as in sharding_util.cpp by assuming the partition id part is always of length 5.